cmd/controller/main.go

```diff
@@ -68,22 +88,80 @@ func main() {
 	flag.Parse()
 	logf.SetLogger(logf.ZapLogger(false))
 
-	// Read the logging config and setup a logger.
-	cm := getLoggingConfigOrDie()
-
-	config, err := logging.NewConfigFromMap(cm, logconfig.Controller)
-	if err != nil {
-		log.Fatalf("Error parsing logging configuration: %v", err)
-	}
-	logger, atomicLevel := logging.NewLoggerFromConfig(config, logconfig.Controller)
+	logger, atomicLevel := setupLogger()
 	defer logger.Sync()
 	logger = logger.With(zap.String(logkey.ControllerType, logconfig.Controller))
 
-	logger.Info("Starting the controller")
-
 	// set up signals so we handle the first shutdown signal gracefully
 	stopCh := signals.SetupSignalHandler()
 
+	go startPkgController(stopCh, logger, atomicLevel)
+	go startControllerRuntime(stopCh, logger, atomicLevel)
+}
+
+func startPkgController(stopCh <-chan struct{}, logger *zap.SugaredLogger, atomicLevel zap.AtomicLevel) {
+	logger = logger.With(zap.String("controller/impl", "pkg"))
+	logger.Info("Starting the controller")
+
+	cfg, err := clientcmd.BuildConfigFromFlags(*masterURL, *kubeconfig)
+	if err != nil {
+		logger.Fatalw("Error building kubeconfig", zap.Error(err))
+	}
+
+	const numControllers = 1
+	cfg.QPS = numControllers * rest.DefaultQPS
+	cfg.Burst = numControllers * rest.DefaultBurst
+	opt := reconciler.NewOptionsOrDie(cfg, logger, stopCh)
+
+	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(opt.KubeClientSet, opt.ResyncPeriod)
+	eventingInformerFactory := informers.NewSharedInformerFactory(opt.EventingClientSet, opt.ResyncPeriod)
+
+	subscriptionInformer := eventingInformerFactory.Eventing().V1alpha1().Subscriptions()
+	//deploymentInformer := kubeInformerFactory.Apps().V1().Deployments()
+	//coreServiceInformer := kubeInformerFactory.Core().V1().Services()
+	configMapInformer := kubeInformerFactory.Core().V1().ConfigMaps()
+
+	// Build all of our controllers, with the clients constructed above.
+	// Add new controllers to this array.
+	controllers := []*kncontroller.Impl{
+		subscription.NewController(
+			opt,
+			subscriptionInformer,
+		),
+	}
+	if len(controllers) != numControllers {
+		logger.Fatalf("Number of controllers and QPS settings mismatch: %d != %d", len(controllers), numControllers)
+	}
+
+	// Watch the logging config map and dynamically update logging levels.
+	opt.ConfigMapWatcher.Watch(logging.ConfigMapName(), logging.UpdateLevelFromConfigMap(logger, atomicLevel, component))
```

Comment:
this uses logging.ConfigMapName() but L177 uses logconfig.ConfigMapName() is that intentional? If so, perhaps add comments about that. Just seems goofy.

Reply:
The two impls are split between the CR way and pkg way. The other path will be deleted when it is removed